src/Colors.js

```diff
@@ -30,9 +30,33 @@ const COLORS = {
 
 const colorValues = Object.values(COLORS);
 
-const randomColor = function() {
-    const colorIndex = Math.floor(Math.random() * colorValues.length);    
-    return colorValues[colorIndex];
+const randomColor = function(exclusionList=[]) {
+    const filteredList = exclusionList.length ? filterList(colorValues, exclusionList) : colorValues;
+    let colorIndex = -1;
+    while (colorIndex < 0 ) {
+        colorIndex = Math.floor(Math.random() * filteredList.length);
+    }
+    return filteredList[colorIndex];
+};
+
+const filterList = (colorValues, exclusionList) => {
+    if (colorValues.length === exclusionList.length) {
+        return [COLORS.WHITE];
+    }
+
+    return colorValues.filter(colorVal => {
```

Comment:
Couldn't you also compare keys here instead of values? Each key is already unique since colors is an object, and comparing the key would probably be more performant than comparing each member of each value

Reply:
that's a much better plan, updated